version: '3.7'

networks:
    smartvuv_net:
        external: true

services:
    traefik:
        image: traefik:v2.3
        command:
            - "--log.level=DEBUG"
            - "--api.insecure=true"
            - "--providers.docker=true"
            - "--providers.docker.exposedbydefault=false"
            - "--entrypoints.http.address=:80"
            - "--entrypoints.https.address=:443"       
        restart: always
        security_opt:
            - no-new-privileges:true
        ports:
            - "80:80"
            - "443:443"
        networks:
            - smartvuv_net
        volumes:
            - /var/run/docker.sock:/var/run/docker.sock:ro
            - ./tools/traefik.yml:/etc/traefik/traefik.yml:ro
            - ./tools/config.yml:/etc/traefik/config.yml:ro
            - ./certs:/etc/certs:ro
            - ./tools/logs/traefik.log:/etc/traefik/traefik.log:rw
        labels:
            - "traefik.enable=true"
            - "traefik.http.routers.traefik=true"    
            - "traefik.http.routers.api.service=api@internal"
            - "traefik.http.routers.traefik.entrypoints=https"            